### PR TITLE
Application Directory Structure

### PR DESCRIPTION
**Pull request by yugandhargangu**
1. Bettle track application directory structure.
2. Power point file of directory
3. Gradle script files and sub modules.
4. Spring boot application configuration

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+before_install: 
+  - cd application/bettle/
+  - chmod +x gradlew
+
+script: "./gradlew clean build"
+
+jdk:
+
+  - oraclejdk8
```
Comment (shashankitmaster) [inline .travis.yml:12]: Please use openjdk for java instead of oracle.

File: application/bettle/.project [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (shashankitmaster) [inline application/bettle/.project]: Remove this file and above `.classpath` . add `.gitignore` file

File: application/bettle/bettle-entity/build.gradle [hunk 3]
```diff
@@ -0,0 +1,9 @@
+apply plugin: "java"
```
Comment (shashankitmaster) [inline application/bettle/bettle-entity/build.gradle:1]: @ranjeet1515 : Add lombok plugin and reduce the code of setter and getter to annotation. 

Refer: https://projectlombok.org/mavenrepo/

File: application/bettle/bettle-jpa/src/main/java/com/bettle/track/jpa/repos/BaseRepository.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.NoRepositoryBean;
+import org.springframework.data.repository.query.Param;
+
+@NoRepositoryBean
+public interface BaseRepository<T extends AbstractParentEntity> extends CrudRepository<T, Long>, JpaRepository<T, Long> {
+
+    @Override
+    @Query("UPDATE #{#entityName} e SET e.active_flag = 0 WHERE e.id = :id")
```
Comment (shashankitmaster) [inline application/bettle/bettle-jpa/src/main/java/com/bettle/track/jpa/repos/BaseRepository.java]: Can we move queries to resources folder and make use of annotation with fetches the query from classpath.

**Review by shashankitmaster — CHANGES_REQUESTED** [review]

**Comment by shashankitmaster** [comment]
Change destination branch to `release` branch

File: .travis.yml (hunk 1 above)
Comment (yugandhargangu) [reply .travis.yml:12]: travis-ci doesn't have openjdk8.
https://github.com/travis-ci/travis-ci/issues/6483

File: application/bettle/bettle-jpa/src/main/java/com/bettle/track/jpa/repos/BaseRepository.java (hunk 4 above)
Comment (yugandhargangu) [reply application/bettle/bettle-jpa/src/main/java/com/bettle/track/jpa/repos/BaseRepository.java]: Yes. This part will be done in jpa specific branch but not in this branch. So I removed this class.

File: application/bettle/.project (hunk 2 above)
Comment (yugandhargangu) [reply application/bettle/.project]: Done.

File: application/bettle/bettle-entity/build.gradle (hunk 3 above)
Comment (yugandhargangu) [reply application/bettle/bettle-entity/build.gradle:1]: Done

File: .travis.yml (hunk 1 above)
Comment (shashankitmaster) [reply .travis.yml:12]: I got your point but I think Oracle will soon licence JDK use.

File: application/bettle/bettle-common/src/main/java/com/bettle/track/enums/CustomFieldType.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+     * @param messageKey message key
+     * @param required   True if Values are mandatory
+     */
+    CustomFieldType(int id, String messageKey, boolean required) {
+        this.id = id;
+        this.messageKey = messageKey;
+        this.required = required;
+    }
+
+    public int getId() {
```
Comment (shashankitmaster) [inline application/bettle/bettle-common/src/main/java/com/bettle/track/enums/CustomFieldType.java:41]: @yugandhargangu : Please get rid of getter and setter as we using lombok plugin for that.
 
Please try to do that for all classes

**Review by shashankitmaster — CHANGES_REQUESTED** [review]

File: application/bettle/bettle-jpa/src/main/java/com/bettle/track/jpa/repos/BaseRepository.java (hunk 4 above)
Comment (shashankitmaster) [reply application/bettle/bettle-jpa/src/main/java/com/bettle/track/jpa/repos/BaseRepository.java]: @yugandhargangu : Got it

**Comment by yugandhargangu** [comment]
Updated with bettle-repo and bettle-service modules.

File: application/bettle/bettle-common/src/main/java/com/bettle/track/enums/CustomFieldType.java (hunk 5 above)
Comment (shashankitmaster) [reply application/bettle/bettle-common/src/main/java/com/bettle/track/enums/CustomFieldType.java:41]: Okay. I am sorry I thought those are classes. I am approving.

**Review by shashankitmaster — APPROVED** [review]
Please merge ur branch

**Pull request merged**